Bugzilla.pm

```diff
@@ -767,6 +768,23 @@ sub elastic {
     $class->process_cache->{elastic} //= Bugzilla::Elastic->new();
 }
 
+sub check_rate_limit {
+    my ($class, $name, $id) = @_;
+    my $params = Bugzilla->params;
+    if ($params->{rate_limit_active}) {
+        my $rules = decode_json($params->{rate_limit_rules});
+        my $limit = $rules->{$name};
+        unless ($limit) {
+             warn "no rules for $name!";
+             return 0;
+        }
+        if (Bugzilla->memcached->should_rate_limit("$name:$id", @$limit)) {
+            Bugzilla->audit("[rate_limit] $id exceeds rate limit $name: " . join("/", @$limit));
```

Comment:
Just to cover here -- in the previous review glob wanted to add the remote_ip here, to protect against the future when $id might not be the remote ip address. We're specifically *not* doing this now because it will just make the log noisier. I'll either add it when we need it, or when we have structured logging. So this part is fine.